machineconfiguration/v1/types.go

```diff
@@ -0,0 +1,439 @@
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	kubeletconfigv1beta1 "k8s.io/kubelet/config/v1beta1"
+)
+
+// TODO: Move MCOConfig to api/operator/v1
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// MCOConfig describes configuration for MachineConfigOperator.
+type MCOConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +required
+	Spec MCOConfigSpec `json:"spec"`
+}
+
+// MCOConfigSpec is the spec for MCOConfig resource.
+type MCOConfigSpec struct {
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// MCOConfigList is a list of MCOConfig resources
+type MCOConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []MCOConfig `json:"items"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ControllerConfig describes configuration for MachineConfigController.
+// This is currently only used to drive the machineconfigs generated by the TemplateController.
+type ControllerConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +required
+	Spec ControllerConfigSpec `json:"spec"`
+	// +optional
+	Status ControllerConfigStatus `json:"status"`
+}
+
+// ControllerConfigSpec is the spec for ControllerConfig resource.
+type ControllerConfigSpec struct {
+	// clusterDNSIP is the cluster DNS IP address
+	ClusterDNSIP string `json:"clusterDNSIP"`
+
+	// cloudProviderConfig is the configuration for the given cloud provider
+	CloudProviderConfig string `json:"cloudProviderConfig"`
+
+	// TODO: Use PlatformType instead of string
+
+	// platform specifies the OpenShift platform, e.g. "libvirt", "openstack", "baremetal", "aws", or "none"
+	Platform string `json:"platform"`
+
+	// etcdDiscoveryDomain specifies the etcd discovery domain
+	EtcdDiscoveryDomain string `json:"etcdDiscoveryDomain"`
+
+	// TODO: Investigate using string for CA data
```

Comment:
Afaiu v1 has already been released, ableit from the mco repo, and not this one. Can I still change this then?

Reply:
same goes for the KubeletConfig type below

Reply:
As we talked about: the use of RawExtensions instead of an explicit kubelet config type will not change the wire format. So that's possible and should be done.